### PR TITLE
issue 21 Add AdamW to the optimization search space

### PR DESCRIPTION
**Pull request by cooperlab**
Newer tf.keras.optimizers incorporate exponential moving averaging of weights.

Using this requires bumping the TF version to 2.11. Also, many existing optimizers are moved temporarily to tf.keras.optimizers.experimental.

### CODE REVIEW

**Comment by cooperlab** [comment]
Note - the steps/iterations in `ema_overwrite_frequency` refers to batches or application of gradients. So this parameter determines whether averaging should be applied after so many batches (`int`), or only once per epoch (`None`).

**Comment by cooperlab** [comment]
This is a complete draft that adds moving averaging for all previously supported optimizers. The example notebook runs.

**Review by lawrence-chillrud — APPROVED** [review]
All looked great - no comments from me. Making a note that exploring conditional searches with ray tune will avoid situations in which `use_ema = False` but ray wants to fine tune e.g. `ema_momentum` anyway. We can explore conditional searches in a later PR.
